lib/logflare_web/open_telemetry_sampler.ex

```diff
@@ -0,0 +1,67 @@
+defmodule LogflareWeb.OpenTelemetrySampler do
+  require OpenTelemetry.Tracer, as: Tracer
+  @behaviour :otel_sampler
```

Comment:
Should check how this is implemented in case we can overload it.

Otherwise probably good for now.

Reply:
Read through the src for it, it uses the generated trace id to filter based on the probability distribution (i think it a uniform dist), if its in the upper bound tail end of the distribution then it will decide. It is is a pure function so cost is low, plus this is decided on span emission so it is quite safe.